portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,23 @@ async function getMessageFromServer() {
   const quote = await response.text();
   document.getElementById('message-container').innerHTML = quote;
 }
+
+function login(url){
+  window.location = url;
+}
+
+function logout(url){
+  window.location = url;
+}
+
+async function checkLoginStatus(){
+  const response = await fetch('/login');
+  const status = await response.text();
+  
+  if(status.localeCompare("/_ah/login?continue=%2Findex.html")){
```

Comment:
comparing strings like this is risky because the underlying login/logout URL may change and then this would break.  If you send the status down explicitly via the JSON object you can check it more easily